src/server/pfs/server/source.go

```diff
@@ -31,13 +31,14 @@ type source struct {
 
 // NewSource creates a Source which emits FileInfos with the information from commit, and the entries return from fileSet.
 func NewSource(commitInfo *pfs.CommitInfo, fs fileset.FileSet, opts ...SourceOption) Source {
+	validatedFS := fileset.NewPathValidator(fs)
```

Comment:
nit, it is preferable to overwrite `fs` rather than define a new variable (i.e. `validatedFS`). This is more maintainable because we may add a new fileset wrapper between this and where we first use it. This is similar to why creating a subcontext usually involves overwriting the existing `ctx`.